src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_ndjson.rs

```diff
@@ -108,10 +111,22 @@ impl InputFormatTextBase for InputFormatNDJson {
                         batch.offset + start,
                         row_info,
                     );
-                    return Err(ErrorCode::BadBytes(msg));
+                    if builder.ctx.on_error_mode == OnErrorMode::Continue {
+                        columns.iter_mut().for_each(|c| {
+                            // check if parts of columns inserted data, if so, pop it.
+                            if c.len() > num_rows {
+                                c.pop_data_value().expect("must success");
+                            }
+                        });
+                        start = *end;
+                        continue;
```

Comment:
@b41sh @BohuTANG In Snowflake, when reading JSON encounter an error, they pop the wrong row, and skip all remaining data, should we follow their approach?

Reply:
We can follow it, Is this likely a simple way?

Reply:
After some thought, I think we can skip the error line and continue reading the rest of the data, because ndjson is new line delimited, not simple json format.

Reply:
skipping bad rows is common.

1.  each format need to handle it by itself.
2. need some limit to give up a file,  maybe the FormatOption is bad or the whole file is bad, processing a whole file result in slow in time and lots of error log.

we can investigate more systems like Spark.

Reply:
I like this advice, currently, we follow the [Snowflake way](https://docs.snowflake.com/en/sql-reference/sql/copy-into-table.html#copy-options-copyoptions). I'll do that in future PRs.

Reply:
LGTM

Reply:
a drawback is: we need other work to make the file upload atomic by file.
we can now abandon the remaining. but not those rows before.
for TSV/NDJson which can be load in parallel, we even can not clearly say "only the first N rows is loaded"

Reply:
related: #9336